.gitlab/includes/nvhpc23_9_pipeline.yml

```diff
@@ -12,12 +12,12 @@ include:
   variables:
     SPACK_ARCH: linux-ubuntu22.04-haswell
     COMPILER: gcc@12.1.0
-    NVHPC_COMPILER: nvhpc@23.9
+    NVHPC_COMPILER: nvhpc@24.3
```

Comment:
Todo: if we keep the version bump, change the pipeline name. Currently the version bump doesn't change anything (i.e. does not get rid of build failures), so I might simply revert this.

Reply:
NVHPC version changed back as it doesn't make a difference. I have bumped the CUDA version used in the NVHPC CI configuration to 12.4 though. It seems like NVHPC with a standalone CUDA version older than 12 fails to compile stdexec, and it seems related to how host/device code is detected. Specifically, https://github.com/NVIDIA/stdexec/blob/ab774d5f84ea4d4e9253e06e82b45ed4b8a8e016/include/stdexec/execution.hpp#L462 ends up expanding to incliding a `__trap` in host code (it's only available on the device) and compilation fails. The function in stdexec that contains `STDEXEC_TERMINATE` used to be a template function that doesn't get instantiatiated in regular use (it's only meant for debugging) so this never came up earlier. Now it's a non-template member function that fails to compile right away.

Since this isn't an issue with newer versions of CUDA I'm not going to make any special workarounds, besides bumping the CUDA version in CI. I'm still trying to make a small reproducer and will open an issue if I get one.